internal/services/cdn/cdn_endpoint_custom_domain_resource.go

```diff
@@ -58,6 +67,73 @@ func resourceArmCdnEndpointCustomDomain() *pluginsdk.Resource {
 				ForceNew:     true,
 				ValidateFunc: validation.StringIsNotEmpty,
 			},
+			"cdn_managed_https": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"certificate_type": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(cdn.CertificateTypeShared),
+								string(cdn.CertificateTypeDedicated),
+							}, false),
+						},
+						"protocol_type": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(cdn.ProtocolTypeServerNameIndication),
+								string(cdn.ProtocolTypeIPBased),
+							}, false),
+						},
+					},
+				},
+				ConflictsWith: []string{"user_managed_https"},
+			},
+
+			"user_managed_https": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"key_vault_id": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: keyvaultValidate.VaultID,
+						},
+						"secret_name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: keyvaultValidate.NestedItemName,
+						},
+						"secret_version": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+				},
+				ConflictsWith: []string{"cdn_managed_https"},
+			},
+		},
+		CustomizeDiff: func(ctx context.Context, diff *pluginsdk.ResourceDiff, _ interface{}) error {
+			if settings, ok := diff.GetOk("cdn_managed_https"); ok {
+				settings := settings.([]interface{})[0].(map[string]interface{})
```

Comment:
FWIK, in case the properties inside the block are not all optional (i.e. at least one required), then if the returned value from `GetOk` is not the default value, then it is guaranteed to contain at least one element.

Anyway, it is always safe to length check for this :)